From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Tariq Bashir <120014322+t-bashir-bs@users.noreply.github.com>
Date: Fri, 3 May 2024 10:25:41 +0100
Subject: OS-16623 add setopacity to wayland toplevel window

This patch adds a setopacity method to the wayland_toplevel_window class to
allow the opacity of the window to be set.

diff --git a/ui/ozone/platform/wayland/host/wayland_toplevel_window.cc b/ui/ozone/platform/wayland/host/wayland_toplevel_window.cc
index 62e7711232d1ed3247a8c52c5f93cad62973c378..dff73038098e12d068e2991de04a2d0e532b26b4 100644
--- a/ui/ozone/platform/wayland/host/wayland_toplevel_window.cc
+++ b/ui/ozone/platform/wayland/host/wayland_toplevel_window.cc
@@ -339,6 +339,11 @@ void WaylandToplevelWindow::SizeConstraintsChanged() {
   SetSizeConstraints();
 }
 
+void WaylandToplevelWindow::SetOpacity(float opacity) {
+  root_surface()->set_opacity(opacity);
+  root_surface()->set_blending(true);
+}
+
 void WaylandToplevelWindow::SetZOrderLevel(ZOrderLevel order) {
   // TODO(crbug.com/374244479): Linux/Wayland doesn't support zorder level.
   // Consider complete removal of that.
diff --git a/ui/ozone/platform/wayland/host/wayland_toplevel_window.h b/ui/ozone/platform/wayland/host/wayland_toplevel_window.h
index 19cee35df2d6ad9545ee0da43c3f169a4112a8f4..92fb79ffa6c7e749f4fd61a4c6440607b2036749 100644
--- a/ui/ozone/platform/wayland/host/wayland_toplevel_window.h
+++ b/ui/ozone/platform/wayland/host/wayland_toplevel_window.h
@@ -103,6 +103,7 @@ class WaylandToplevelWindow : public WaylandWindow,
   void SetWindowIcons(const gfx::ImageSkia& window_icon,
                       const gfx::ImageSkia& app_icon) override;
   void SizeConstraintsChanged() override;
+  void SetOpacity(float opacity) override;
   // `SetZOrderLevel()` must be called on `z_order_` in
   // `SetUpShellIntegration()`.
   void SetZOrderLevel(ZOrderLevel order) override;
